client/src/components/common/Debounce.tsx

```diff
@@ -0,0 +1,43 @@
+import {useCallback, useEffect, useState} from "react";
+type UseDebounceProps<T> = {
```

Comment:
could just move all this into its own folder and use a types.ts file to store all the types